### ISSUE TITLE
Fatal error when using reflection on type having config with typeReachable condition.

### ISSUE THREAD

**Issue by christophstrobl** [oracle/graal#4418]
**Describe the issue**
We are using reflection to extract method metadata for an interface hierarchy. To keep this externally provided reflection configuration small we use conditions which seems to be problematic at this point.

The actual user code is written in Koltin (`ReservationRepository`) and extends a `JpaRepository` from Spring Data. 

```kotlin
@Entity
data class Reservation(var name: String, @Id @GeneratedValue var id: Int? = null)

interface ReservationRepository : JpaRepository<Reservation, Int> {
	fun findByName(name: String): Reservation?
}
```

To satisfy reflection requirements of Kotlin, the type hierarchy for `ReservationRepository` involving `JpaRepository` needs to be enabled for reflection. The app works fine as long as no `typeReachable` condition is present.

```json
{
  "name": "org.springframework.data.jpa.repository.JpaRepository",
  "queryAllPublicMethods": true
}
```

Once a condition is added, we see a `Fatal error` when running the application which is trying to gather annotation information on methods.

```json
{
  "name": "org.springframework.data.jpa.repository.JpaRepository",
  "condition": {
    "typeReachable": "javax.persistence.EntityManager"
  },
  "queryAllPublicMethods": true
}    
```

**Steps to reproduce the issue**
A reproducer (unfortunately neither simple, nor minimal) can be found in this [branch](https://github.com/christophstrobl/spring-graalvm-native/tree/issue/1536).
1. there's a `./build.sh` in the root folder that will build _spring-native_ along with the hint (via `DataJpaHints`) adding the reflection configuration for the `JpaRepository` mentioned above.
2. goto _samples/data-jpa-kotlin_ and build it with maven `mvn -Pnative clean package`.
3. run the app `./target/data-jpa-kotlin`.  --> you should see the error printed to the console

Removing the condition in [DataJpaHints.java#L45](https://github.com/christophstrobl/spring-graalvm-native/blob/issue/1536/spring-native-configuration/src/main/java/org/springframework/data/jpa/repository/config/DataJpaHints.java#L45) and rebuilding via steps 1-3 from above will allow the app to start without the error.

**Describe GraalVM and your environment:**
- GraalVM: GraalVM CE 22.0.0.2 (build 11.0.14+9-jvmci-22.0-b05)
- OS: macOS 10.15.7 (19H1824)

**More details**

_Generated Configuration:_
The spring-native code generation will put native image config files in `target/generated-runtime-sources/spring-aot/src/main/resources/META-INF/native-image/org.springframework.aot/spring-aot`

_Error_
```
Fatal error

Printing Instructions (ip=0x0000000107a5b340):
  0x0000000107a5b320: 0x00 0x41 0xc7 0x87 0xe4 0x00 0x00 0x00 0xfe 0xfe 0xfe 0x7e 0x48 0x8b 0x7c 0x24
  0x0000000107a5b330: 0x20 0x48 0x8b 0x74 0x24 0x10 0x48 0x8b 0x54 0x24 0x18 0xe8 0xa0 0xfd 0xff 0xff
  0x0000000107a5b340: 0x90 0xcc 0xcc 0xcc 0xcc 0xcc 0xcc 0xcc 0xcc 0xcc 0xcc 0xcc 0xcc 0xcc 0xcc 0xcc
  0x0000000107a5b350: 0x48 0x83 0xec 0x18 0x49 0x3b 0x67 0x08 0x0f 0x86 0xb0 0x00 0x00 0x00 0x48 0x8b

Top of stack (sp=0x00007ffee8517770):
  0x00007ffee8517770: 0x0000000111f0e838 0x0000000117b57880
  0x00007ffee8517780: 0x000000010e600000 0x000000010e600000
  0x00007ffee8517790: 0x0000000108662378 0x0000000107bee8d4
  0x00007ffee85177a0: 0x0000000111f0dda0 0x0000000108662378
  0x00007ffee85177b0: 0x0000000110c2c4d8 0x00000001086665b3
  0x00007ffee85177c0: 0x000000010e600000 0x000000010a210b80
  0x00007ffee85177d0: 0x0000000117ae92d0 0x000000010850551a
  0x00007ffee85177e0: 0x0000000117ae7738 0x000000010a521647
  0x00007ffee85177f0: 0x0000000001248468 0x0000000117b57880
  0x00007ffee8517800: 0x0000000117aeac20 0x000000000000000d
  0x00007ffee8517810: 0x000000010f112d68 0x000000010a21100a
  0x00007ffee8517820: 0x000000000f112d68 0x000000010a21079e
  0x00007ffee8517830: 0x0000000111f0f878 0x000000010f1129e8
  0x00007ffee8517840: 0x0000000117b56e30 0x0000000117b56f90
  0x00007ffee8517850: 0x000000010f112d68 0x000000000000000a
  0x00007ffee8517860: 0x0000000a0000000b 0x0000000e0000000a
  0x00007ffee8517870: 0x0000000200000001 0x000000010a213338
  0x00007ffee8517880: 0x0000000000000001 0x000000010a210813
  0x00007ffee8517890: 0x0000000117b4e4a0 0x000000010f112d68
  0x00007ffee85178a0: 0x0000000117b52b48 0x0000000117b54ad0
  0x00007ffee85178b0: 0x0000000117b54b10 0x000000010f0d8958
  0x00007ffee85178c0: 0x0000002300000024 0x000000010f1129e8
  0x00007ffee85178d0: 0x0000000100000001 0x000000010a213512
  0x00007ffee85178e0: 0x9970cd4cd606f224 0x0000000108a4bdf4
  0x00007ffee85178f0: 0x0000000117b56f70 0x000000010f1129e8
  0x00007ffee8517900: 0x0000000117b52b48 0x0000000117b54ad0
  0x00007ffee8517910: 0x0000000117b54b10 0x000000010f0d8958
  0x00007ffee8517920: 0x0000000117b4e970 0x000000020f0c6978
  0x00007ffee8517930: 0x0000000100000000 0x000000010a213512
  0x00007ffee8517940: 0x000000010feb9a50 0x000000010e600000
  0x00007ffee8517950: 0x0000000117b54b28 0x000000010f0c6978
  0x00007ffee8517960: 0x0000000117b52b48 0x0000000117b54ad0

Top frame info:
  TotalFrameSize in CodeInfoTable 48

Threads:
  0x00007f99a6706540 STATUS_IN_NATIVE (ALLOW_SAFEPOINT) "HikariPool-1 housekeeper" - 0x000000011646d6d8, daemon, stack(0x000070000b775000,0x000070000b7f5000)
  0x00007f99964048c0 STATUS_IN_NATIVE (ALLOW_SAFEPOINT) "boundedElastic-evictor-1" - 0x00000001141918f8, daemon, stack(0x000070000b6f2000,0x000070000b772000)
  0x00007f99a8005040 STATUS_IN_NATIVE (ALLOW_SAFEPOINT) "Signal Dispatcher" - 0x0000000111f03410, daemon, stack(0x000070000b569000,0x000070000b5e9000)
  0x00007f99a6605b40 STATUS_IN_JAVA (PREVENT_VM_FROM_REACHING_SAFEPOINT) "main" - 0x0000000110128828, stack(0x00007ffee7d19000,0x00007ffee8519000)

VM thread locals for the failing thread 0x00007f99a6605b40:
  0 (8 bytes): JNIThreadLocalEnvironment.jniFunctions = (bytes)
    0x00007f99a6605b40: 0x000000010f7a1df8
  8 (8 bytes): StackOverflowCheckImpl.stackBoundaryTL = (Word) 1 (0x0000000000000001)
  16 (4 bytes): Safepoint.safepointRequested = (int) 2139803752 (0x7f8ad068)
  20 (4 bytes): StatusSupport.statusTL = (int) 1 (0x00000001)
  24 (32 bytes): ThreadLocalAllocation.regularTLAB = (bytes)
    0x00007f99a6605b58: 0x0000000117b00000 0x0000000117c00000
    0x00007f99a6605b68: 0x0000000117b57be0 0x0000000114300000
  56 (8 bytes): JavaThreads.currentThread = (Object) java.lang.Thread (0x0000000110128828)
  64 (8 bytes): JavaFrameAnchors.lastAnchor = (Word) 0 (0x0000000000000000)
  72 (8 bytes): ExceptionUnwind.currentException = (Object) null
  80 (8 bytes): IdentityHashCodeSupport.hashCodeGeneratorTL = (Object) java.util.SplittableRandom (0x0000000111f1b860)
  88 (8 bytes): IsolatedCompileClient.currentClient = (Object) null
  96 (8 bytes): IsolatedCompileContext.currentContext = (Object) null
  104 (8 bytes): JNIObjectHandles.handles = (Object) com.oracle.svm.core.handles.ThreadLocalHandles (0x0000000111f01180)
  112 (8 bytes): JNIThreadLocalPendingException.pendingException = (Object) null
  120 (8 bytes): JNIThreadLocalPinnedObjects.pinnedObjectsListHead = (Object) null
  128 (8 bytes): JNIThreadOwnedMonitors.ownedMonitors = (Object) null
  136 (8 bytes): NoAllocationVerifier.openVerifiers = (Object) null
  144 (8 bytes): ThreadingSupportImpl.activeTimer = (Object) null
  152 (8 bytes): SubstrateDiagnostics.threadOnlyAttachedForCrashHandler = (bytes)
    0x00007f99a6605bd8: 0x0000000000000000
  160 (8 bytes): VMThreads.IsolateTL = (Word) 4536139776 (0x000000010e600000)
  168 (8 bytes): VMThreads.OSThreadHandleTL = (Word) 4635590080 (0x00000001144d7dc0)
  176 (8 bytes): VMThreads.OSThreadIdTL = (Word) 4635590080 (0x00000001144d7dc0)
  184 (8 bytes): VMThreads.StackBase = (Word) 140732796080128 (0x00007ffee8519000)
  192 (8 bytes): VMThreads.StackEnd = (Word) 140732787691520 (0x00007ffee7d19000)
  200 (8 bytes): VMThreads.nextTL = (Word) 0 (0x0000000000000000)
  208 (8 bytes): VMThreads.unalignedIsolateThreadMemoryTL = (Word) 140297898056480 (0x00007f99a6605b20)
  216 (4 bytes): ActionOnExitSafepointSupport.actionTL = (int) 0 (0x00000000)
  220 (4 bytes): ActionOnTransitionToJavaSupport.actionTL = (int) 0 (0x00000000)
  224 (4 bytes): ImplicitExceptions.implicitExceptionsAreFatal = (int) 0 (0x00000000)
  228 (4 bytes): StackOverflowCheckImpl.yellowZoneStateTL = (int) 2130640638 (0x7efefefe)
  232 (4 bytes): StatusSupport.safepointBehaviorTL = (int) 1 (0x00000001)
  236 (4 bytes): ThreadingSupportImpl.currentPauseDepth = (int) 0 (0x00000000)

No VMOperation in progress

The 15 most recent VM operation status changes (oldest first):

Counters:

Java frame anchors for the failing thread 0x00007f99a6605b40:
  No anchors

Stacktrace for the failing thread 0x00007f99a6605b40:
  SP 0x00007ffee8517770 IP 0x0000000107a5b340  [image code] com.oracle.svm.core.jdk.VMErrorSubstitutions.shutdown(VMErrorSubstitutions.java:116)
  SP 0x00007ffee8517770 IP 0x0000000107a5b340  [image code] com.oracle.svm.core.jdk.VMErrorSubstitutions.shouldNotReachHere(VMErrorSubstitutions.java:109)
  SP 0x00007ffee85177a0 IP 0x0000000107bee8d4  [image code] com.oracle.svm.core.util.VMError.shouldNotReachHere(VMError.java:58)
  SP 0x00007ffee85177b0 IP 0x0000000108662378  [image code] java.lang.reflect.Executable.declaredAnnotations(Executable.java:135)
  SP 0x00007ffee85177b0 IP 0x0000000108662378  [image code] java.lang.reflect.Executable.getDeclaredAnnotations(Executable.java:590)
  SP 0x00007ffee85177c0 IP 0x00000001086665b3  [image code] java.lang.reflect.Method.getDeclaredAnnotations(Method.java:703)
  SP 0x00007ffee85177d0 IP 0x000000010a210b80  [image code] org.springframework.core.annotation.AnnotationsScanner.getDeclaredAnnotations(AnnotationsScanner.java:454)
  SP 0x00007ffee8517830 IP 0x000000010a21079e  [image code] org.springframework.core.annotation.AnnotationsScanner.getBaseTypeMethods(AnnotationsScanner.java:348)
  SP 0x00007ffee8517880 IP 0x000000010a213338  [image code] org.springframework.core.annotation.AnnotationsScanner.processMethodHierarchy(AnnotationsScanner.java:294)
  SP 0x00007ffee85178e0 IP 0x000000010a213512  [image code] org.springframework.core.annotation.AnnotationsScanner.processMethodHierarchy(AnnotationsScanner.java:313)
  SP 0x00007ffee8517940 IP 0x000000010a213512  [image code] org.springframework.core.annotation.AnnotationsScanner.processMethodHierarchy(AnnotationsScanner.java:313)
  SP 0x00007ffee85179a0 IP 0x000000010a212b7c  [image code] org.springframework.core.annotation.AnnotationsScanner.processMethod(AnnotationsScanner.java:250)
  SP 0x00007ffee85179e0 IP 0x000000010a2118ec  [image code] org.springframework.core.annotation.AnnotationsScanner.process(AnnotationsScanner.java:89)
  SP 0x00007ffee85179f0 IP 0x000000010a21385b  [image code] org.springframework.core.annotation.AnnotationsScanner.scan(AnnotationsScanner.java:77)
  SP 0x00007ffee8517a10 IP 0x000000010a229e8d  [image code] org.springframework.core.annotation.TypeMappedAnnotations.scan(TypeMappedAnnotations.java:242)
  SP 0x00007ffee8517a30 IP 0x000000010a228ee7  [image code] org.springframework.core.annotation.TypeMappedAnnotations.get(TypeMappedAnnotations.java:149)
  SP 0x00007ffee8517a60 IP 0x000000010a20d3a4  [image code] org.springframework.core.annotation.TypeMappedAnnotations.get(TypeMappedAnnotations.java:131)
  SP 0x00007ffee8517a60 IP 0x000000010a20d3a4  [image code] org.springframework.core.annotation.AnnotationUtils.findAnnotation(AnnotationUtils.java:531)
  SP 0x00007ffee8517a90 IP 0x000000010a32e984  [image code] org.springframework.data.repository.core.support.DefaultRepositoryInformation.isQueryAnnotationPresentOn(DefaultRepositoryInformation.java:175)
  SP 0x00007ffee8517a90 IP 0x000000010a32e984  [image code] org.springframework.data.repository.core.support.DefaultRepositoryInformation.isQueryMethodCandidate(DefaultRepositoryInformation.java:163)
  SP 0x00007ffee8517ab0 IP 0x000000010a32dd0e  [image code] org.springframework.data.repository.core.support.DefaultRepositoryInformation.getQueryMethods(DefaultRepositoryInformation.java:146)
  SP 0x00007ffee8517b00 IP 0x000000010a32e7d8  [image code] org.springframework.data.repository.core.support.DefaultRepositoryInformation.isQueryMethod(DefaultRepositoryInformation.java:193)
  SP 0x00007ffee8517b20 IP 0x000000010a2ea13c  [image code] org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor.lambda$new$0(CrudMethodMetadataPostProcessor.java:114)
  SP 0x00007ffee8517b30 IP 0x000000010a2e90f9  [image code] org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor$$Lambda$679139ca40af1c7a6ae7b364fe21e4897230686c.matches(Unknown Source)
  SP 0x00007ffee8517b40 IP 0x000000010a53b4dd  [image code] org.springframework.util.ReflectionUtils.doWithMethods(ReflectionUtils.java:363)
  SP 0x00007ffee8517ba0 IP 0x000000010a2e9524  [image code] org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor.<init>(CrudMethodMetadataPostProcessor.java:113)
  SP 0x00007ffee8517bd0 IP 0x000000010a2eac9f  [image code] org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor.postProcess(CrudMethodMetadataPostProcessor.java:78)
  SP 0x00007ffee8517c00 IP 0x000000010a346136  [image code] org.springframework.data.repository.core.support.RepositoryFactorySupport.lambda$getRepository$3(RepositoryFactorySupport.java:352)
  SP 0x00007ffee8517c10 IP 0x000000010a34135c  [image code] org.springframework.data.repository.core.support.RepositoryFactorySupport$$Lambda$0efd40d70601882b96468f43084134f0d945d269.accept(Unknown Source)
  SP 0x00007ffee8517c20 IP 0x000000010884db7a  [image code] java.util.ArrayList.forEach(ArrayList.java:1541)
  SP 0x00007ffee8517c70 IP 0x000000010a3449f0  [image code] org.springframework.data.repository.core.support.RepositoryFactorySupport.getRepository(RepositoryFactorySupport.java:347)
  SP 0x00007ffee8517cf0 IP 0x000000010a341008  [image code] org.springframework.data.repository.core.support.RepositoryFactoryBeanSupport.lambda$afterPropertiesSet$5(RepositoryFactoryBeanSupport.java:322)
  SP 0x00007ffee8517d00 IP 0x000000010a33fc22  [image code] org.springframework.data.repository.core.support.RepositoryFactoryBeanSupport$$Lambda$f7fc054a567834720cc8eade7c22a5be2695243c.get(Unknown Source)
  SP 0x00007ffee8517d10 IP 0x000000010a382823  [image code] org.springframework.data.util.Lazy.getNullable(Lazy.java:230)
  SP 0x00007ffee8517d30 IP 0x000000010a3825e3  [image code] org.springframework.data.util.Lazy.get(Lazy.java:114)
  SP 0x00007ffee8517d50 IP 0x000000010a3406bb  [image code] org.springframework.data.repository.core.support.RepositoryFactoryBeanSupport.afterPropertiesSet(RepositoryFactoryBeanSupport.java:328)
  SP 0x00007ffee8517d90 IP 0x000000010a2f1268  [image code] org.springframework.data.jpa.repository.support.JpaRepositoryFactoryBean.afterPropertiesSet(JpaRepositoryFactoryBean.java:144)
  SP 0x00007ffee8517db0 IP 0x000000010a00853d  [image code] org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863)
  SP 0x00007ffee8517e00 IP 0x000000010a007c24  [image code] org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800)
  SP 0x00007ffee8517e30 IP 0x000000010a003b3f  [image code] org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620)
  SP 0x00007ffee8517ea0 IP 0x000000010a00234d  [image code] org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
  SP 0x00007ffee8517f00 IP 0x000000010a018823  [image code] org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
  SP 0x00007ffee8517f30 IP 0x000000010a00e528  [image code] org.springframework.beans.factory.support.AbstractBeanFactory$$Lambda$2c40051508af481f3e2b011afbd5800cdeae9fd3.getObject(Unknown Source)
  SP 0x00007ffee8517f40 IP 0x000000010a0483a2  [image code] org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
  SP 0x00007ffee8517fc0 IP 0x000000010a011605  [image code] org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
  SP 0x00007ffee8518060 IP 0x0000000109ffb6e2  [image code] org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
  SP 0x00007ffee8518060 IP 0x0000000109ffb6e2  [image code] org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
  SP 0x00007ffee8518070 IP 0x000000010a039b7d  [image code] org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1389)
  SP 0x00007ffee85180f0 IP 0x000000010a0424ee  [image code] org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1309)
  SP 0x00007ffee8518130 IP 0x0000000109f5bc6d  [image code] org.springframework.aot.beans.factory.InjectedConstructionResolver.lambda$resolve$0(InjectedConstructionResolver.java:83)
  SP 0x00007ffee8518140 IP 0x0000000109f5b9f3  [image code] org.springframework.aot.beans.factory.InjectedConstructionResolver$$Lambda$2e00153b7e05d644436a095c0da3c269e26a6a7d.get(Unknown Source)
  SP 0x00007ffee8518150 IP 0x0000000109f5db1b  [image code] org.springframework.aot.beans.factory.InjectedConstructionResolver.resolveDependency(InjectedConstructionResolver.java:97)
  SP 0x00007ffee8518180 IP 0x0000000109f5c733  [image code] org.springframework.aot.beans.factory.InjectedConstructionResolver.resolve(InjectedConstructionResolver.java:83)
  SP 0x00007ffee8518230 IP 0x000000010ac8500d  [image code] org.springframework.aot.beans.factory.InjectedElementResolver.resolve(InjectedElementResolver.java:35)
  SP 0x00007ffee8518230 IP 0x000000010ac8500d  [image code] org.springframework.aot.beans.factory.InjectedElementResolver.create(InjectedElementResolver.java:66)
  SP 0x00007ffee8518250 IP 0x0000000109f58f19  [image code] org.springframework.aot.beans.factory.BeanDefinitionRegistrar$BeanInstanceContext.create(BeanDefinitionRegistrar.java:211)
  SP 0x00007ffee8518270 IP 0x0000000109f4e841  [image code] org.springframework.aot.ContextBootstrapInitializer.lambda$initialize$2(ContextBootstrapInitializer.java:172)
  SP 0x00007ffee8518290 IP 0x0000000109f356c2  [image code] org.springframework.aot.ContextBootstrapInitializer$$Lambda$69eb1cda6626bbd13955778dd9e04a64ffb4b62c.applyWithException(Unknown Source)
  SP 0x00007ffee85182a0 IP 0x000000010ac84af2  [image code] org.springframework.aot.beans.factory.BeanDefinitionRegistrar$ThrowableFunction.apply(BeanDefinitionRegistrar.java:294)
  SP 0x00007ffee85182d0 IP 0x0000000109f5aaf2  [image code] org.springframework.aot.beans.factory.BeanDefinitionRegistrar.lambda$instanceSupplier$0(BeanDefinitionRegistrar.java:115)
  SP 0x00007ffee85182f0 IP 0x0000000109f58aa2  [image code] org.springframework.aot.beans.factory.BeanDefinitionRegistrar$$Lambda$a3df89430a3ba7d397d0053371a7f0bbfc99ecb2.get(Unknown Source)
  SP 0x00007ffee8518300 IP 0x000000010a008835  [image code] org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.obtainFromSupplier(AbstractAutowireCapableBeanFactory.java:1249)
  SP 0x00007ffee8518340 IP 0x000000010a002cf1  [image code] org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1191)
  SP 0x00007ffee8518390 IP 0x000000010a003beb  [image code] org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:582)
  SP 0x00007ffee8518400 IP 0x000000010a00234d  [image code] org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
  SP 0x00007ffee8518460 IP 0x000000010a018823  [image code] org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
  SP 0x00007ffee8518490 IP 0x000000010a00e528  [image code] org.springframework.beans.factory.support.AbstractBeanFactory$$Lambda$2c40051508af481f3e2b011afbd5800cdeae9fd3.getObject(Unknown Source)
  SP 0x00007ffee85184a0 IP 0x000000010a0483a2  [image code] org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
  SP 0x00007ffee8518520 IP 0x000000010a011605  [image code] org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
  SP 0x00007ffee85185c0 IP 0x000000010a03f775  [image code] org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
  SP 0x00007ffee85185c0 IP 0x000000010a03f775  [image code] org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:953)
  SP 0x00007ffee8518620 IP 0x000000010a1c509d  [image code] org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:918)
  SP 0x00007ffee8518660 IP 0x000000010a1c8dad  [image code] org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
  SP 0x00007ffee85186c0 IP 0x000000010a1a638b  [image code] org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.refresh(ReactiveWebServerApplicationContext.java:64)
  SP 0x00007ffee85186e0 IP 0x000000010a06c67c  [image code] org.springframework.boot.SpringApplication.refresh(SpringApplication.java:740)
  SP 0x00007ffee85186e0 IP 0x000000010a06c67c  [image code] org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:415)
  SP 0x00007ffee8518700 IP 0x000000010a06cb76  [image code] org.springframework.boot.SpringApplication.run(SpringApplication.java:303)
  SP 0x00007ffee8518760 IP 0x000000010a06d070  [image code] org.springframework.boot.SpringApplication.run(SpringApplication.java:1312)
  SP 0x00007ffee8518780 IP 0x000000010775464d  [image code] org.springframework.boot.SpringApplication.run(SpringApplication.java:1301)
  SP 0x00007ffee8518780 IP 0x000000010775464d  [image code] com.example.webflux.ApplicationKt.main(Application.kt:76)
  SP 0x00007ffee85187a0 IP 0x0000000107a01885  [image code] com.oracle.svm.core.JavaMainWrapper.runCore(JavaMainWrapper.java:150)
  SP 0x00007ffee85187a0 IP 0x0000000107a01885  [image code] com.oracle.svm.core.JavaMainWrapper.run(JavaMainWrapper.java:186)
  SP 0x00007ffee85187d0 IP 0x0000000107a2615d  [image code] com.oracle.svm.core.code.IsolateEnterStub.JavaMainWrapper_run_5087f5482cc9a6abc971913ece43acb471d2631b(IsolateEnterStub.java:0)

VM mutexes:
  mutex "thread" is unlocked.
  mutex "referencePendingList" is unlocked.
  mutex "mainVMOperationControlWorkQueue" is unlocked.

Heap settings and statistics:
  Supports isolates: true
  Heap base: 0x000000010e600000
  Object reference size: 8
  Aligned chunk size: 1048576
  Incremental collections: 0
  Complete collections: 0

Native image heap boundaries:
  ReadOnly Primitives: 0x000000010e701028 - 0x000000010f062c98
  ReadOnly References: 0x000000010f062c98 - 0x000000010f79a168
  ReadOnly Relocatables: 0x000000010f79b000 - 0x000000010fd400e0
  Writable Primitives: 0x000000010fd41000 - 0x000000010febf608
  Writable References: 0x000000010febf608 - 0x00000001110a8998
  Writable Huge: 0x0000000111100030 - 0x00000001111ce7c8
  ReadOnly Huge: 0x00000001111cf030 - 0x0000000111edd560

Heap:
  Young generation:
    Eden:
      edenSpace:
        aligned: 1048576/1 unaligned: 0/0
        aligned chunks:
          0x0000000116600000 (0x0000000116601028-0x0000000116603190)
    Survivors:
      Survivor-1 From:
        aligned: 0/0 unaligned: 0/0
      Survivor-1 To:
        aligned: 0/0 unaligned: 0/0
      Survivor-2 From:
        aligned: 0/0 unaligned: 0/0
      Survivor-2 To:
        aligned: 0/0 unaligned: 0/0
      Survivor-3 From:
        aligned: 0/0 unaligned: 0/0
      Survivor-3 To:
        aligned: 0/0 unaligned: 0/0
      Survivor-4 From:
        aligned: 0/0 unaligned: 0/0
      Survivor-4 To:
        aligned: 0/0 unaligned: 0/0
      Survivor-5 From:
        aligned: 0/0 unaligned: 0/0
      Survivor-5 To:
        aligned: 0/0 unaligned: 0/0
      Survivor-6 From:
        aligned: 0/0 unaligned: 0/0
      Survivor-6 To:
        aligned: 0/0 unaligned: 0/0
      Survivor-7 From:
        aligned: 0/0 unaligned: 0/0
      Survivor-7 To:
        aligned: 0/0 unaligned: 0/0
      Survivor-8 From:
        aligned: 0/0 unaligned: 0/0
      Survivor-8 To:
        aligned: 0/0 unaligned: 0/0
      Survivor-9 From:
        aligned: 0/0 unaligned: 0/0
      Survivor-9 To:
        aligned: 0/0 unaligned: 0/0
      Survivor-10 From:
        aligned: 0/0 unaligned: 0/0
      Survivor-10 To:
        aligned: 0/0 unaligned: 0/0
      Survivor-11 From:
        aligned: 0/0 unaligned: 0/0
      Survivor-11 To:
        aligned: 0/0 unaligned: 0/0
      Survivor-12 From:
        aligned: 0/0 unaligned: 0/0
      Survivor-12 To:
        aligned: 0/0 unaligned: 0/0
      Survivor-13 From:
        aligned: 0/0 unaligned: 0/0
      Survivor-13 To:
        aligned: 0/0 unaligned: 0/0
      Survivor-14 From:
        aligned: 0/0 unaligned: 0/0
      Survivor-14 To:
        aligned: 0/0 unaligned: 0/0
      Survivor-15 From:
        aligned: 0/0 unaligned: 0/0
      Survivor-15 To:
        aligned: 0/0 unaligned: 0/0
  Old generation:
    oldFromSpace:
      aligned: 0/0 unaligned: 0/0
    oldToSpace:
      aligned: 0/0 unaligned: 0/0

  Unused:
    aligned: 0/0

Fatal error
```

Related to: spring-projects-experimental/spring-native#1536

**Comment by sdeleuze** [comment]
Please add the `spring` label to this issue.

@gradinac I think I remember you fixed some `queryMethod` handling in the past, but I can't remember the details. Could be related.

**Comment by oubidar-Abderrahim** [comment]
Hi, Thank you for reporting this, we will take a look into it and get back to you

**Comment by sdeleuze** [comment]
@oubidar-Abderrahim Sounds like this kind of issue has been fixed with 22.1-dev for at least some use case we have, worth to check and close this if confirmed.

**Comment by oubidar-Abderrahim** [comment]
tested and could not reproduce with 22.2
